lib_board_support/api/boards/xk_audio_316_mc_ab/board.h

```diff
@@ -3,56 +3,127 @@
 #pragma once
 
+/**
+ * \addtogroup xk_audio_316_mc_ab xk_audio_316_mc_ab
+ *
+ * The common defines for using lib_board_support.
+ * @{
+ */
+
+#include "i2c.h"
+
+
 #if __XC__
-#define BS_PORT port
+#define BS_PORT_T port
 #else
 #include <xcore/port.h>
-#define BS_PORT port_t
+#define BS_PORT_T port_t
 #endif
 
 #include <xccompat.h>
 #include "i2c.h"
 
-/* I2C interface ports */
-extern BS_PORT p_scl;
-extern BS_PORT p_sda;
+/** I2C interface ports */
+extern BS_PORT_T p_scl;
+extern BS_PORT_T p_sda;
 
 
+/** Type of clock to be instantiated. This may be a fixed clock using the application PLL,
+ * an adjustable clock using the CS2100 external PLL or and adjustable clock using application
+ * PLL.
+ */
 typedef enum {
     CLK_FIXED,
     CLK_CS2100,
     CLK_PLL
 } xk_audio_316_mc_ab_mclk_modes_t;
 
+/** Formats supported by the DAC and ADC. Either I2S using multiple data lines or TDM
+ * supporting multi-channel using a single data line.
+ */
 typedef enum {
     AUD_316_PCM_FORMAT_I2S,
     AUD_316_PCM_FORMAT_TDM
 } xk_audio_316_mc_ab_pcm_format_t;
 
+/** @struct xk_audio_316_mc_ab_config_t
+ *  @brief Configuration struct type for setting the hardware profile.
+ *  @var xk_audio_316_mc_ab_config_t::clk_mode
+ *  See xk_audio_316_mc_ab_mclk_modes_t for available clock mode options.
+ *  @var xk_audio_316_mc_ab_config_t::dac_is_clock_master
+ *  Boolean setting for whether the DAC or the xcore.ai is I2S clock master. Set to 0 to make the xcore.ai master.
+ *  @var xk_audio_316_mc_ab_config_t::default_mclk
+ *  Nominal clock frequency in MHz. Standard rates are supported between 11.2896 MHz and 49.152 MHz.
+ *  @var xk_audio_316_mc_ab_config_t::pll_sync_freq
+ *  When the CLK_CS2100 is used, this defines the nominal reference clock frequency for multiplication by the PLL.
+ *  This value is ignored when the CS2100 is not used.
+ *  *  @var xk_audio_316_mc_ab_config_t::pcm_format
+ *  See xk_audio_316_mc_ab_pcm_format_t for available data frame options.
+ *  @var xk_audio_316_mc_ab_config_t::i2s_n_bits
+ *  Number of bits per data frame in I2S.
+ *  @var xk_audio_316_mc_ab_config_t::i2s_chans_per_frame
+ *  This defines the number of audio channels per frame (a frame is a complete cycle of FSYNC or LRCLK).
+ * 
+ */
 typedef struct {
     xk_audio_316_mc_ab_mclk_modes_t clk_mode;
     char dac_is_clock_master; // bool
-
-    // fixed clock config
     unsigned default_mclk;
-
-    // cs2100 clock config
     unsigned pll_sync_freq;
-
     xk_audio_316_mc_ab_pcm_format_t pcm_format;
     unsigned i2s_n_bits;
-
     unsigned i2s_chans_per_frame;
 } xk_audio_316_mc_ab_config_t;
 
-/// Start an i2s master thread which uses the DAC pins
+
+/** Starts an I2C master task. Must be started from tile[0] *after* xk_audio_316_mc_ab_board_setup() and *before* and tile[1] HW calls.
+ *
+ *  \param   i2c        client side of I2C master interface connection.
+ *  \param   config     Reference to the xk_audio_316_mc_ab_config_t configuration struct.
```

Comment:
this param doesn't exist

Reply:
Ha - just spotted another similar issue too. Fixed.